### ISSUE TITLE
Wrong altitude display - add adjustable offset to displayed value

### ISSUE THREAD

**Issue by Andrea777-ai** [direc85/sailfish-gpsinfo#37]
With this tweak one can bring the altitude display to the correct value, adding or subtracting an adjustable value to the altitude data from GPS. 

To add a slider for adjusting the offset, I inserted this into `pages/FirstPage.qml`, beginning at line 185:

```
            SectionHeader {                                                                               
                anchors.left: parent.left                                                                
                anchors.leftMargin: Theme.paddingMedium * 2                                  
                text: "Altitude Offset"                                           
            }                                                                          
                                                                                                  
            Slider {                                                                             
                minimumValue: -100                                                              
                maximumValue: 100                                                       
                stepSize: 1                                                            
                value: settings.altitudeOffset
                valueText: value + "m"                                                           
                width: parent.width                                                          
                onReleased: settings.altitudeOffset = value                             
            }                                                                          

```
to add or subtract the value for display, I inserted the new variable `settings.altitudeOffset` in line 95 and 97 to calculate the displayed value - code snipped from line 89:
```
            InfoField {
                label: qsTr("Altitude")
                visible: settings.showAltitudeApp
                value: {
                    if (providers.position.position.altitudeValid) {
                        if (settings.units === "MET") {
                            return locationFormatter.roundToDecimal(providers.position.position.coordinate.altitude - settings.altitudeOffset, 2) + " m"
                        } else {
                            return locationFormatter.roundToDecimal((providers.position.position.coordinate.altitude - settings.altitudeOffset) * 3.2808399, 2) + " ft"
                        }
                    }
                    return "-"
                }
            }

```
Similar in `pages/CoverPage.qml` , from line 56:
```
        InfoField {
            label: qsTr("Altitude")
            visible: settings.showAltitudeCover
            fontpixelSize: Theme.fontSizeMedium
            value: {
                if (providers.position.position.altitudeValid) {
                    if (settings.units == "MET") {
//                        return locationFormatter.roundToDecimal(providers.position.position.coordinate.altitude, 2) + " m"
                        return locationFormatter.roundToDecimal(providers.position.position.coordinate.altitude - settings.altitudeOffset, 2) + " m"
                    } else {
//                        return locationFormatter.roundToDecimal(providers.position.position.coordinate.altitude * 3.2808399, 2) + " ft"
                        return locationFormatter.roundToDecimal((providers.position.position.coordinate.altitude - settings.altitudeOffset) * 3.2808399, 2) + " ft"
                    }
                }
                return "-"
            }
        }

```
tldr:
on the beach I had an altitude display of 57m, now I can set it to zero or 1m. The necessary offset value depends from the place on earth, where the user is.
